core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/service/DiagnosticsTriggerService.java

```diff
@@ -71,9 +81,12 @@ public FlowIdentifier startDiagnosticsCollection(BaseDiagnosticsCollectionReques
                 && StringUtils.isNotBlank(stack.getCluster().getBlueprint().getStackVersion())) {
             clusterVersion = stack.getCluster().getBlueprint().getStackVersion();
         }
+        String accountId = Crn.fromString(stack.getResourceCrn()).getAccountId();
+        boolean useDbusCnameEndpoint = entitlementService.useDataBusCNameEndpointEnabled(INTERNAL_ACTOR_CRN, accountId);
+        String databusEndpoint = dataBusEndpointProvider.getDataBusEndpoint(telemetry.getDatabusEndpoint(), useDbusCnameEndpoint);
```

Comment:
This logic is very similar to the logic in TelemetryDecorator and in ClouderaManagerMgmtTelemetryService, might worth to make an own method for it somewhere

Reply:
common has no auth-connector dependency, first i tried to put these kind of code together in auth-connector but @lnardai did not want that (I understand) - so i cannot include Entitlementservice in DataBusEndpointProvider

Reply:
ohh i see, makes sense